crates/services/consensus_module/poa/src/signer.rs

```diff
@@ -173,4 +198,24 @@ mod tests {
             .is_available())
         };
     }
+
+    #[test]
+    fn public_key_address() {
+        assert_eq!(SignMode::Unavailable.public_key_address().unwrap(), None);
+
+        let mut rng = StdRng::seed_from_u64(2322);
+        let secret_key = SecretKey::random(&mut rng);
+        let public_key_address = SignMode::Key(Secret::new(secret_key.into()))
+            .public_key_address()
+            .unwrap();
+
+        assert!(public_key_address.is_some());
+
+        #[cfg(feature = "aws-kms")]
+        {
+            // Ideally here we want to retrieve the secp256k1 public key from a
+            // key arn and a kms client here.
+            todo!()
```

Comment:
I was looking at the mockall crate (https://docs.aws.amazon.com/sdk-for-rust/latest/dg/testing.html) to try to mock KMS, so that ideally I can generate a key arn and try to extract the public key address from it. I wanted to check that this is okay before proceeding.